code/game/objects/structures/vehicles/vehicle.dm

```diff
@@ -95,16 +96,56 @@
 			to_chat(user, "Need more welding fuel!")
 			return
 	else if(istype(W, /obj/item/key))
-		if(keytype)
-			to_chat(user, "Hold \the [W] in one of your hands while you drive \the [src].")
+		if(!heldkey)
+			if(keytype && mykey == W)
+				if(prob(95) && user.drop_item(W, src))
+					to_chat(user, "<span class='notice'>You insert \the [W] to \the [src]'s ignition and turn it.</span>")
+					user.visible_message("<span class='notice'>\The [src]'s engine roars to life!</span>")
+					playsound(get_turf(src), "sound/items/screwdriver.ogg", 10, 1) // find a better sound later
+					src.heldkey = W
+				else
+					to_chat(user, "<span class='warning'>You try to insert \the [W] to \the [src]'s ignition but you miss the slot!</span>")
+			else
+				if(keytype)
+					to_chat(user, "<span class='warning'>\The [W] doesn't fit into \the [src]'s ignition.</span>")
+				else
+					to_chat(user, "<span class='notice'>You don't need a key.</span>")
 		else
-			to_chat(user, "You don't need a key.")
+			to_chat(user, "<span class='notice'>\the [src]'s already has \the [heldkey] in it.</span>")
+	else if(isscrewdriver(W) && !heldkey)
+		var/mob/living/carbon/human/H = user
+		to_chat(user, "<span class='warning'>You jam \the [W] into \the [src]'s ignition and feel like a genius as you try turning it!</span>")
+		H.adjustBrainLoss(10)
+
+/obj/structure/bed/chair/vehicle/verb/remove_key()
+	set name = "Remove Key"
+	set category = "Object"
+	set src in oview(1)
+
+	if(heldkey && !usr.incapacitated() && Adjacent(usr) && usr.dexterity_check())
```

Comment:
Using `usr` like this is generally a bad idea, because there's no guarantee `usr` exists.
You should at least wrap all the parts that access members of `usr` in an `if(usr)`

Reply:
Not sure what you mean by "there's no guarantee usr exists." Technically something must exist for it to preform the verb or am I crazy?

Reply:
I'm just going to cut the verb out completely and put it all in the attack_hand in place of it calling the verb.

Reply:
Verbs can be called just like procs
In fact, the only difference between `/verb` and `/proc` is the default value of the `category` setting
Unless I forgot something
Either way, verbs are just a special case of procs, and as such can be called just as well by non-mobs